api/v2/api.go

```diff
@@ -224,7 +224,7 @@ func (api *API) getReceiversHandler(params receiver_ops.GetReceiversParams) midd
 
 	receivers := make([]*open_api_models.Receiver, 0, len(api.alertmanagerConfig.Receivers))
 	for _, r := range api.alertmanagerConfig.Receivers {
-		receivers = append(receivers, &open_api_models.Receiver{Name: &r.Name})
```

Comment:
This tripped me up for a moment, but I get it now.

The problem here is that we're storing a pointer to `r` where `r` will be the last value that's used. Instead of changing the API spec, I think we should do:

```golang
receivers = append(receivers, &open_api_models.Receiver{Name: &api.alertmanagerConfig.Receivers[i].Name})
```

Which defefintely solves the problem.

Can we also please add some tests to this endpoint?

Reply:
I applied your suggested changes and add a test.